### PR TITLE
Multi App SSO for Android

### PR DESCRIPTION
**Pull request by wmathurin**
Update to Android IDP template (goes with https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2393)

### CODE REVIEW

File: AndroidIDPTemplate/app/AndroidManifest.xml [hunk 1]
```diff
@@ -58,30 +58,5 @@
         </activity>
         -->
 
-		<!-- IDP account picker activity -->
```
Comment (wmathurin) [inline AndroidIDPTemplate/app/AndroidManifest.xml:61]: No manifest change needed in the app.

File: AndroidIDPTemplate/app/src/com/salesforce/samples/salesforceandroididptemplateapp/MainActivity.kt [hunk 2]
```diff
@@ -160,32 +160,33 @@ class MainActivity : SalesforceActivity() {
...
+        val spAppPackageName = when (appName) {
+            MOBILE_SYNC_EXPLORER -> MOBILE_SYNC_EXPLORER_PACKAGE
+            REST_EXPLORER -> REST_EXPLORER_PACKAGE
+            ACCOUNT_EDITOR -> ACCOUNT_EDITOR_PACKAGE
+            else -> null
+        }
+        Log.d(TAG, "Launching SP app ${appName} with package ${spAppPackageName}")
+        if (spAppPackageName != null) {
+            SalesforceSDKManager.getInstance().idpManager?.let { idpManager ->
+                idpManager.kickOffIDPInitiatedLoginFlow(this, spAppPackageName,
```
Comment (wmathurin) [inline AndroidIDPTemplate/app/src/com/salesforce/samples/salesforceandroididptemplateapp/MainActivity.kt:172]: Basic way an app could show the status updates to the user (toasts). But other approaches are possible (progress bar etc).

File: AndroidIDPTemplate/app/src/com/salesforce/samples/salesforceandroididptemplateapp/MainApplication.kt [hunk 3]
```diff
@@ -43,12 +44,30 @@ class MainApplication : Application() {
...
         MobileSyncSDKManager.getInstance().registerUsedAppFeature(FEATURE_APP_USES_KOTLIN)
 
-        /*
-         * Uncomment the following line to enable IDP login flow. This will allow the user to
-         * either authenticate using the current app or use a designated IDP app for login.
-         * Replace 'idpAppURIScheme' with the URI scheme of the IDP app meant to be used.
-         */
-        // MobileSyncSDKManager.getInstance().idpAppURIScheme = idpAppURIScheme
+        // Setting app as IDP for these allowed SP apps
+        MobileSyncSDKManager.getInstance().setAllowedSPApps(listOf(
```
Comment (wmathurin) [inline AndroidIDPTemplate/app/src/com/salesforce/samples/salesforceandroididptemplateapp/MainApplication.kt:48]: Now the IDP app needs to have more details about the SP apps it want to work with.

**Review by brandonpage — APPROVED** [review]
I believe it is possible to automate all of the IDP flows through UIAutomator2 on Android (and XCUITest on iOS).  Obviously not high priority but I'll put it on the backlog for when things slow down a bit.

**Pull request merged**